central_v2/src/pages/Confirmation.tsx

```diff
@@ -0,0 +1,207 @@
+    color: '#02215F',
+    fontFamily: 'Poppins, sans-serif',
+    fontWeight: 700,
+    fontSize: '24px',
+    lineHeight: '36px',
+    textAlign: 'center',
+}));
+
+const EnterText = styled(Typography)(({ theme }) => ({
+    fontWeight: 400,
+    fontSize: '16px',
+    lineHeight: '20px',
+    fontFamily: 'Rubik, sans-serif',
+    color: '#02215F',
+}));
+
+const CodeandResendContainer = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '8px',
+}));
+
+const UserCodeTextBoxesContainer = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'center',
+    gap: '8px',
+}));
+
+const UserCodeTextBoxes = styled(TextField)(({ theme }) => ({
+    width: '40px',
+    height: '54px',
+    borderRadius: '8px',
+    border: '2px solid #FFFFFF',
+    color: '#FFFFFF',
+    textAlign: 'center',
+    input: {
+        textAlign: 'center',
+    },
+}));
+
+const ResendCodeText = styled(Typography)(({ theme }) => ({
+    fontFamily: 'Rubik, sans-serif',
+    fontWeight: 400,
+    fontSize: '16px',
+    color: '#02215F',
+    textDecoration: 'underline',
+    textAlign: 'center',
+}));
+
+const VerifyBox = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'center',
+}));
+
+// Props interface
+interface ConfirmationProps {
+    schoolEmail?: string;
+    frontImage?: File | null;
+    backImage?: File | null;
+    handlerImageUpload?: (file: File) => Promise<any>;
+    password?: string
+}
+
+// Use function declaration for the component
+function Confirmation({ schoolEmail = '', frontImage, backImage, handlerImageUpload, password}: ConfirmationProps) {
+    const [code, setCode] = useState(Array(6).fill(''));
+    const apiClients = useTSAPIClientsContext(APIClientsContext);
+
+    const inputRefs = useRef<Array<HTMLInputElement | null>>([]); // Refs for each input box
+
+    const handleChange = (value: string, index: number) => {
+        if (!/^[0-9]*$/.test(value)) return; // Only allow numeric input
+        const newCode = [...code];
+        newCode[index] = value;
+        setCode(newCode);
+
+        // Automatically move to the next input box if a character is entered
+        if (value && index < 5) {
+            inputRefs.current[index + 1]?.focus();
+        }
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent, index: number) => {
+        if (e.key === 'Backspace' && !code[index] && index > 0) {
+            inputRefs.current[index - 1]?.focus();
+        }
+    };
+
+    const handleSubmit = async () => {
+        const fullCode = code.join('');
+        if (fullCode.length < 6) {
+            alert('Please enter all 6 digits of the confirmation code.');
+            return;
+        }
+        try {
+            await apiClients.auth.awsConfirmSignUp(schoolEmail, fullCode);
+            console.log('Confirmation successful!');
+
+        } catch (error) {
+            console.error('Error confirming sign-up:', error);
+        }
+
+        try {
```

Comment:
given that each of these apiClient calls is sequential (waiting for the first to conclude before the second happens) it's good you're using await. 

That being said, this might be an example of when to use the `.then()` syntax, as it would nest these all down, clean up the `try/catch` formatting and make it super clear that they were all going to fire one after the other.